### PR TITLE
RDG013 sample code

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

File: fundamentals/aot/diagnostics/Rdg13/Program.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}
+// </snippet_1>
+#elif RDG13F
+// <snippet_1f>
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+builder.Services.AddScoped<ITodoService, TodoService>();
```
Comment (Rick-Anderson) [inline fundamentals/aot/diagnostics/Rdg13/Program.cs:43]: @captainsafia I'm registering ITodoService here but at run time I get `System.InvalidOperationException:` 'No service for type 'ITodoService' has been registered.'

cc @timdeschryver

File: fundamentals/aot/diagnostics/Rdg13/Program.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.TypeInfoResolverChain.Insert(0,
+                                       AppJsonSerializerContext.Default);
+});
+
+
+var app = builder.Build();
+
+//app.MapGet("/v1/todos",
```
Comment (Rick-Anderson) [inline fundamentals/aot/diagnostics/Rdg13/Program.cs:54]: @captainsafia  this is the endpoint that throws.

File: fundamentals/aot/diagnostics/Rdg13/Program.cs (hunk 1 above)
Comment (timdeschryver) [reply fundamentals/aot/diagnostics/Rdg13/Program.cs:43]: Hi Rick, you can fix this by registering the service using `AddKeyedScoped`.

https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8#keyed-di-services

```cs
builder.Services.AddKeyedScoped<ITodoService, TodoService>("primary");
```

**Comment by Rick-Anderson** [comment]
@captainsafia  fixed in #203

**Pull request closed without merging**